server/src/main/java/org/opensearch/cluster/metadata/Metadata.java

```diff
@@ -1425,6 +1452,48 @@ public Builder generateClusterUuidIfNeeded() {
         }
 
         public Metadata build() {
+            TimeValue buildStartTime = TimeValue.timeValueMillis(System.nanoTime());
+            DataStreamMetadata dataStreamMetadata = (DataStreamMetadata) this.customs.get(DataStreamMetadata.TYPE);
+            boolean recomputeRequired = indices.equals(indicesPreviousState) == false
```

Comment:
this doesn't account for change in aliases which also require recomputing the indicesLookup

Reply:
> this doesn't account for change in aliases which also require recomputing the indicesLookup

Aliases information is present in `indices`: https://github.com/opensearch-project/OpenSearch/pull/7853#discussion_r1222178818